misc/path_utils.c

```diff
@@ -41,17 +41,18 @@
 
 char *mp_basename(const char *path)
 {
+    if (mp_is_url(bstr0(path)))
+        return (char *)path;
```

Comment:
as an aside: why does this function not return a `const char*`?

Reply:
Apparently it did until 962eec0440 with no explanation for the change. Anyway, I can change it after this PR.